### PR TITLE
3.3 - Sass (and some css confusion)

### PR DESCRIPTION
**Pull request by easmith2**
~~I'm having trouble with placement of objects (using margins/padding/position/etc). For an example of what I mean, check out how the article title moves around when you resize (because at the moment, it's based on %, not a fixed value) or the misalignment of the horizontal rules when the page is split vertically. Can I meet with either of you (ahem, Roberto or Adam) about how to fix that at some point?~~

Think I've figured it out! (with a lot of help from others, that is). I'm going to work on it a little more and hopefully get the positioning fixed.

### CODE REVIEW

**Comment by rpearce** [comment]
Well done using the SVG. Looks badass

File: sass/_lines.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+.line_gradientLight {
+  width: 100%;
+  border: 0;
+  height: 1px;
+  margin-right: -5px;
+  margin-left: 0;
+  background-image: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 0.35), rgba(255, 255, 255, 0.85), rgba(255, 255, 255, 0.35), rgba(255, 255, 255, 0));
```
Comment (rpearce) [inline sass/_lines.scss]: Awesome. what about on other/older browsers?

File: sass/_lines.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.line_gradientLight {
+  width: 100%;
+  border: 0;
+  height: 1px;
+  margin-right: -5px;
+  margin-left: 0;
```
Comment (rpearce) [inline sass/_lines.scss]: is this line necessary?

File: sass/_lines.scss [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  background-image: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 0.35), rgba(255, 255, 255, 0.85), rgba(255, 255, 255, 0.35), rgba(255, 255, 255, 0));
+  @media all and (max-width: $break-w) {
+    display: none;
+  }
+}
+.line_gradientDark {
+  width: 100%;
```
Comment (rpearce) [inline sass/_lines.scss]: is this line necessary, or would it make more sense for it to be `display: block`?

File: sass/_lines.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  border: 0;
+  height: 1px;
+  margin-right: -5px;
+  margin-left: 0;
```
Comment (rpearce) [inline sass/_lines.scss]: is this line necessary?

File: sass/_lines.scss (hunk 3 above)
Comment (rpearce) [inline sass/_lines.scss]: This isn't wrong, just fyi

File: sass/_mixins.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// @mixin name(argument) {
+//   stuff
+// }
+//
+
+
```
Comment (rpearce) [inline sass/_mixins.scss]: if the above is no longer being used, I'd crush it from the code

File: sass/_mixins.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+@mixin position($top, $right:null, $bottom:null, $left: null) {
+  top: $top;
+  right: $right;
+  bottom: $bottom;
+  left: $left;
+}
+@mixin relative($positions...) {
+  position: relative;
+  @include position($positions...);
```
Comment (rpearce) [inline sass/_mixins.scss]: I don't create a `relative` mixin, for it's a tough thing to control outside of the `top` option.

File: sass/_post.scss [hunk 7]
```diff
@@ -0,0 +1,94 @@
+.post__header {
+  background-image: url(../images/background.jpeg);
+  background-position: 70%, 50%;
+  background-size: cover;
+  @include relative(0, null, 0);
+  padding: 2% 0% 30% 2%;
+  width: 100%;
```
Comment (rpearce) [inline sass/_post.scss]: I don't think this line is necessary

File: sass/_post.scss [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    background-position: 0, 26;
+    background-size: cover;
+    width: 50%;
+    padding: 5% 0% 0% 5%;
+  }
+}
+.post__headerLogo {
+  position: absolute;
+  top: 15px;
+  left: 15px;
```
Comment (rpearce) [inline sass/_post.scss]: maybe use an `absolute` mixin?

File: sass/_post.scss [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+}
+.post__headerNav {
+  position: absolute;
+  top: 35px;
+  left: 15px;
```
Comment (rpearce) [inline sass/_post.scss]: here, too?

File: sass/_post.scss [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    }
+    &:last-of-type:after {
+      content: none;
+    }
+  }
+}
+.post__headerIcon {
+  position: absolute;
+  top: 10px;
+  right: 10px;
```
Comment (rpearce) [inline sass/_post.scss]: and here?

File: sass/_post.scss [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+  border: 1px solid #FFF;
+  border-radius: 50%;
+  width: 40px;
+  height: 40px;
```
Comment (rpearce) [inline sass/_post.scss]: maybe a `size()` mixin might be useful here

File: sass/_post.scss [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+}
+.post__headerIcon img {
```
Comment (rpearce) [inline sass/_post.scss]: if you'd like to nest this within what is on line 42, you can do that

File: sass/_post.scss [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+  width: 40px;
+  height: 40px;
+  margin: 5px 0px 5px 8px;
+  overflow: hidden;
+}
+.post__headerTitle {
+  position: absolute;
+  width: 100%;
+  top: 50px;
+  left: 15px;
```
Comment (rpearce) [inline sass/_post.scss]: same comment as above with `absolute`

File: sass/_post.scss [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+  font-size: 2rem;
+  font-family: Helvetica Narrow, sans-serif;
```
Comment (rpearce) [inline sass/_post.scss]: any fallbacks?

File: sass/_post.scss [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+  height: 40px;
+  margin: 5px 0px 5px 8px;
+  overflow: hidden;
+}
+.post__headerTitle {
+  position: absolute;
+  width: 100%;
+  top: 50px;
+  left: 15px;
+  font-size: 2rem;
```
Comment (rpearce) [inline sass/_post.scss]: :+1:

File: sass/_post.scss [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+  @media all and (max-width: $break-w) {
+    top: 90px;
+    width: auto;
+  }
+}
+.post__content {
+  @include relative(null);
+  width: 100%;
+  margin: 0;
+  padding: 20px 40px 20px 40px;
```
Comment (rpearce) [inline sass/_post.scss]: you don't actually need to repeat yourself here.
* whatever you apply to the top will be applied to the bottom unless the bottom is defined
* whatever you apply to the right will be applied to the left unless the left is defined

**Comment by rpearce** [comment]
groovy!

File: sass/_lines.scss (hunk 1 above)
Comment (easmith2) [inline sass/_lines.scss]: I tested the line appearance in safari and firefox (a-okay), but only with the latest versions. After some light reading on this, I've added some lines that should account for use in Chrome 25+, Safari 6+, iOS 6.1+, Android 4.3+, Firefox 3.6 to 15, and even Opera 11.1 to 12.0 (with a fallback option of a plain background color, should the gradient not be supported).

File: sass/_post.scss (hunk 14 above)
Comment (easmith2) [inline sass/_post.scss]: is using sans-serif as the last option covering that?

**Pull request merged**